escodegen.js

```diff
@@ -1954,10 +1954,11 @@
 
         MetaProperty: function (expr, precedence, flags) {
             var result;
-            result = [];
-            result.push(expr.meta);
-            result.push('.');
-            result.push(expr.property);
+            result = [
+              expr.meta.type === Syntax.Identifier ? expr.meta.name : String(expr.meta),
```

Comment:
Why do we use `String(...)` here? Also, if the properties were just strings before, we should test `typeof expr.meta.type === "string"`, not try to access a nonexistent `type` property.